### PR TITLE
Stormpath Filter after Spring Security

### LINKED ISSUE #892: AuthenticationSuccessEvent no longer published on successful login
With the move to 1.0.2 an `AuthenticationSuccessEvent` is no longer published when a successful login occurs.

Previously I believe the Spring Security `ProviderManager` used to iterate through all the authentication providers and if one was successful it would trigger the publishing of an `AuthenticationSuccessEvent`. See [ProviderManager.java:220](https://github.com/spring-projects/spring-security/blob/f221920a1990f458b26b185c8fbe5b3623dbd8c1/core/src/main/java/org/springframework/security/authentication/ProviderManager.java#L220):

    eventPublisher.publishAuthenticationSuccess(result);

Now with 1.0.2. this logic is not invoked, rather the Stormpath `SpringSecurityResolvedAccountFilter` directly calls `authenticate(...)` on `StormpathAuthenticationProvider` and there is no corresponding event publishing step.

### LINKED ISSUE #915: StormpathFilter must be behind Spring Security Filter
Relates to #892 #1145

### PR DESCRIPTION
**Pull request by mrioan**
For both Spring and Spring Boot

Resolves #892 
Resolves #915

### CODE REVIEW

File: extensions/spring/stormpath-spring-security-webmvc/src/test/groovy/com/stormpath/spring/config/MinimalStormpathSpringSecurityWebMvcConfigurationIT.groovy [hunk 1]
```diff
@@ -243,6 +268,18 @@ class MinimalStormpathSpringSecurityWebMvcConfigurationIT extends AbstractTestNG
...
     }
 
+    /**
+     * @since 1.0.3
+     */
+    @Test
+    void testAuthenticationSuccessEventIsTriggered() {
+        providerManager.setAuthenticationEventPublisher(authenticationEventPublisher)
+        MinimalStormpathSpringSecurityWebMvcTestAppConfig.CustomAuthenticationSuccessEventListener.eventWasTriggered = false
+        mvc.perform(post("/login").accept(MediaType.TEXT_HTML.toString()).contentType(ContentType.TEXT_HTML.toString()).with(csrf()).param("login", account.getEmail()).param("password", password)).andReturn()
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-security-webmvc/src/test/groovy/com/stormpath/spring/config/MinimalStormpathSpringSecurityWebMvcConfigurationIT.groovy]: nice! 👍

**Comment by dogeared** [comment]
@mrioan: code looks solid. However, this introduces 12 failures in the TCK that we need to hunt down.

I ran the TCK against the Spring Security Spring Boot WebMVC and Spring Security Spring WebMVC examples.

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/AccountProviderRequestHandler.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+// stormpath-framework-tck
+/**
+ * @since 1.0.3
+ */
+public interface AccountProviderRequestHandler {
+
+    ProviderAccountRequest getAccountProviderRequest(HttpServletRequest request, Map<String, Object> props);
+    ProviderAccountRequest getAccountProviderRequest(HttpServletRequest request);
+
+    String exchangeGithubCodeForAccessToken(String code, HttpServletRequest request);
```
Comment (mrioan) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/AccountProviderRequestHandler.java]: This seems like a generic interface but it has a reference to a concrete provider like Github. I will analyze if I can think of a solution for this.

Comment (dogeared) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/AccountProviderRequestHandler.java]: Github is the only one of the providers that requires that extra step. Definitely open to any improvements.

**Comment by dogeared** [comment]
Great feedback @mrioan. I was able to refactor the `AccountProviderRequestHandler` interface to get rid of the github specific method altogether.

**Comment by mrioan** [comment]
There are still some pending tasks befor this PR can really be reviewed

**Comment by mrioan** [comment]
This PR is quite out of date now. Closing now.

**Pull request closed without merging**